lib/spack/spack/test/config.py

```diff
@@ -334,10 +335,10 @@ def __init__(self, path):
 
 
 def test_substitute_config_variables(mock_low_high_config, monkeypatch):
-    prefix = spack.paths.prefix.lstrip('/')
+    prefix = spack.paths.prefix.lstrip('/').lstrip('C:\\')
```

Comment:
Using `lstrip` to remove a prefix string is potentially fragile, since it isn't actually removing the exact string `"C:\\"` but rather any combination of characters consisting of any subset the characters in `"C:\\"` (so for example a prefix like `/Crazy/starting/prefix/` would be incorrectly modified).

Also, I'm concerned about removing `C:\` because a call like `canonicalize_path($spack/foo/bar/baz/)` should produce a functional path on windows, and it looks like currently if you have Spack in `C:\a\path\to\spack` that it will return `\a\path\to\spack` - would a user be able to read files from such a path (i.e. with the drive letter removed)?

Reply:
This is no longer necessary.